### PR TITLE
Introduce golangci-lint to keep a clean codebase

### LINKED ISSUE #19: Add a sensible golangcilint configuration for the lambda

### PR DESCRIPTION
**Pull request by Yantrio**
This PR introduces golangci-lint into our CI/CD pipeline for automated code quality checks. By incorporating this tool, we can enforce code consistency and catch issues early in the development process.

I've configured a set of linters based on common Go best practices. However, please note that some rules related to logging have been temporarily disabled. This is a deliberate choice to allow for future improvements in our logging mechanism.

Fixes #19

### CODE REVIEW

**Review by marcinwyszynski — APPROVED** [review]
I sincerely hope you squash :)

**Review by cube2222 — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.54
+          working-directory: src
+          args: --timeout=10m
```
Comment (marcinwyszynski) [inline .github/workflows/lint.yml]: I think that a newline wouldn't hurt here.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
```
Comment (mbialon) [inline .github/workflows/lint.yml]: The test workflow resolves Go version from the `go.mod` file. Do we want to apply the same approach here? 

```
uses: actions/setup-go@v4
with: { go-version-file: 'src/go.mod' }
```

**Review by mbialon — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 2 above)
Comment (Yantrio) [reply .github/workflows/lint.yml]: good catch! yes. I'll introduce this now

**Pull request merged**